Framework/Beamline/src/SpectrumInfo.cpp

```diff
@@ -0,0 +1,32 @@
+#include "MantidBeamline/SpectrumDefinition.h"
+#include "MantidBeamline/SpectrumInfo.h"
+#include "MantidKernel/make_cow.h"
+
+namespace Mantid {
+namespace Beamline {
+
+SpectrumInfo::SpectrumInfo(const size_t numberOfDetectors)
+    : m_spectrumDefinition(Kernel::make_cow<std::vector<SpectrumDefinition>>(
```

Comment:
So the vector of spectrum definitions initially is sized correctly but each `SpectrumDefinition` is empty.